### PR TITLE
HDDS-6579. Add commandline argument for Container Info command.

### PR DESCRIPTION
**Pull request by mukul1987**
## What changes were proposed in this pull request?

Adds a command line option called --addReplicaDetails to print replica related information for container info command

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6579

## How was this patch tested?

Fixed an existing test

### CODE REVIEW

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/container/InfoSubcommand.java [hunk 1]
```diff
@@ -63,6 +63,11 @@ public class InfoSubcommand extends ScmSubcommand {
       description = "Format output as JSON")
   private boolean json;
 
+  @CommandLine.Option(names = { "--addReplicaDetails" },
```
Comment (adoroszlai) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/container/InfoSubcommand.java:66]:
```suggestion
  @CommandLine.Option(names = { "--replicas" },
```

File: hadoop-hdds/tools/src/test/java/org/apache/hadoop/hdds/scm/cli/container/TestInfoSubCommand.java [hunk 2]
```diff
@@ -85,7 +85,7 @@ public void testReplicasIncludedInOutput() throws Exception {
         .thenReturn(getReplicas());
     cmd = new InfoSubcommand();
     CommandLine c = new CommandLine(cmd);
-    c.parseArgs("1");
+    c.parseArgs("1", "--addReplicaDetails");
```
Comment (adoroszlai) [inline hadoop-hdds/tools/src/test/java/org/apache/hadoop/hdds/scm/cli/container/TestInfoSubCommand.java:88]:
```suggestion
    c.parseArgs("1", "--replicas");
```

**Review by adoroszlai — COMMENTED** [review]
Thanks @mukul1987 for working on this.  LGTM.  I suggest simplifying the name of the flag a bit.

**Comment by sodonnel** [comment]
In my opinion this change needs reverted and rethought. When we have 10 such changes to this **human readable** command output, and someone needs all the information are they supposed to pass something like the below?

```
ozone admin container list --a --b --c --d --e --f --g --h --i --j --k ....
```

The user friendly approach is to display all the output. In order to be compatible, we don't guarantee the total output will never change - more we should guarantee information will never be removed, but more information can be added. Otherwise we are overly constraining ourselves for the future. In this case the new information is in a new section of the report and does not alter any existing information. It meets the criteria of "adding and not modifying existing code".

For these "free text" CLI commands. which are generally intended for an administrator to read and use to debug problems, I also believe we should not even be concerned about the formatting of the existing information changing, but I am open to discuss the pros and cons of that. I can see cases where we may want to add some additional information in an existing line of output as the product matures and features are added.

Most commands provide a JSON output too, and it is designed to be consumed by a computer rather than a human. If someone develops a tool to consume this data, they should be consuming the JSON data, which has a well defined parsing approach, and is suited to addition of new fields and data. If the program consuming it is correctly coded, it will not be impacted by new additions, provided existing fields do not move location and are not renamed.

This approach here is not scalable as we simply cannot add flag after flag to for each new piece of information we want to add to each command.

For the record, I am -1 on this change, and I would like it reverted and a discussion in the community to agree a way forward if my arguments here are not sufficient to convince you this is a bad path to continue down.

**Comment by fapifta** [comment]
I agree with Stephen, and I would also like to have this change to be reverted.

container info command is an administrative command, which should by default contain as much information as we can provide, and it should not require CLI options to be added to give one the full picture... This is the comfortable way, as when a human wants to check something it should have all the information in an easy to understand format. What is easy to understand can be a subject for a debate (and the current format is not the best I think), but not presenting an information renders that information to be invisible instead of easy to understand.

If we seem it useful to restrict the amount of information provided, it is better to add options for restricting the amount of information shown about a container, though I don't think there is too much use for that at the moment, as the amount of information about 1 container is not that long so far...

**Pull request merged**